src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -1598,6 +1601,18 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         RsDiagnostic.AwaitOutsideAsyncContext(element, fix).addToHolder(holder)
     }
 
+    private fun checkTypeParameterList(holder: RsAnnotationHolder, path: RsPath) {
+        val resolved = path.reference?.resolve() ?: return
+        val typeParameter = path.contextOfType<RsPathType>()?.owner?.context as? RsTypeParameter ?: return
+        val typeParameterList = (typeParameter.parent as? RsTypeParameterList)?.typeParameterList ?: return
+
+        val parameterIndex = typeParameterList.indexOf(typeParameter).takeIf { it != -1 } ?: return
+        val referencedParameterIndex = typeParameterList.indexOf(resolved).takeIf { it != -1 } ?: return
+        if (parameterIndex < referencedParameterIndex) {
```

Comment:
Seems good! But this code can be further simplified:

```suggestion
        if (resolved in typeParameter.rightSiblings) {
```